tap_googleads/streams.py

```diff
@@ -87,37 +109,69 @@ def gaql(self):
         )
     ).to_dict()
 
-    # Goal of this stream is to send to children stream a dict of
-    # login-customer-id:customer-id to query for all queries downstream
-    def get_records(self, context: Optional[dict]) -> Iterable[Dict[str, Any]]:
-        """Return a generator of row-type dictionary objects.
-
-        Each row emitted should be a dictionary of property names to their values.
+    def validate_response(self, response: Response) -> None:
+        """Catch the customer not enabled error."""
+        if response.status_code == HTTPStatus.FORBIDDEN:
+            errors = next(iter(response.json()["error"].get("details", {}))).get(
+                "errors",
+                [],
+            )
+            if len(errors) == 1:  # make sure this is the only error
+                error = errors[0]
+                if (
+                    error.get("errorCode", {}).get("authorizationError")
+                    == "CUSTOMER_NOT_ENABLED"
+                ):
+                    self.logger.warning(error["message"])
+                    return None
+        return super().validate_response(response)
```

Comment:
@ReubenFrankel  what do you think?

Reply:
Looks alright. That `authorizationError` is not the one I was seeing though. Also, is this scenario not already covered by your change here: https://github.com/Matatika/tap-googleads/blob/b06eb21230fe560a7a1c8a97a7332874740cb908/tap_googleads/streams.py#L161-L166

Reply:
what was the `authorizationError` you had? No this is a different scenario and a different level of customer account. Google ads api is a bit complicated. Maybe its better to just leave the error as is if its too many different variables.

Reply:
> what was the `authorizationError` you had?

https://github.com/Matatika/tap-googleads/pull/45#discussion_r1757219868

> Maybe its better to just leave the error as is if its too many different variables.

Yeah, I think so. I did just try the other customer ID I had as `login_customer_id` and it all seems to work OK, so that's a user error (although could have been clearer). Maybe we are biting more off than we can chew here, and surfacing the API error would be reasonable enough... I think we should keep the logic of skipping customers scoped to `post_process`, if possible - of course, that is contingent on the request completing successfully (still unclear on possible scenarios).

Reply:
I agree!